Backends/Wayland/port.lisp

```diff
@@ -0,0 +1,332 @@
+  (format t "PORT-FORCE-OUTPUT~%")
+  ;; (alx:when-let* ((graft (graft port))
+  ;;                 (mirror (sheet-mirror graft))
+  ;;                 (egl-display (wayland-egl-mirror-display mirror))
+  ;;                 (egl-surface (wayland-egl-mirror-surface mirror)))
+  ;;   (format t "egl swap buffers~%")
+  ;;   (egl:swap-buffers egl-display egl-surface))
+  (map-over-grafts #'%graft-force-output port)
+  (wlc:wl-surface-commit (wayland-port-window port)))
+
+;;; Port event handling
+(defmethod handle-event
+    ((port wayland-port) (event window-configuration-event))
+
+  (let ((sheet (event-sheet event))
+        (native-region (window-event-native-region event)))
+    (format t "window-configuration-event ~S ~S~%" sheet native-region)))
+
+(defmethod port-set-mirror-geometry
+    ((port wayland-port) (sheet mirrored-sheet-mixin) region)
+  (alx:when-let ((mirror (sheet-direct-mirror sheet)))
+    (with-bounding-rectangle* (x1 y1 x2 y2 :width w :height h) region
+      (with-bounding-rectangle* (ox1 ox2 oy1 oy2) (sheet-mirror-geometry sheet)
+        (let ((window (window mirror)))
+          (when (some #'/=
+                      (list x1 y1 x2 y2)
+                      (list ox1 oy1 ox2 oy2))
+            (format t "port-set-mirror-geometry~%")
+            ;TODO: determine if we need CLX's ROUND-COORDINATE
+            (xdg:xdg-surface-set-window-geometry (slot-value port 'surface) x1 y1 w h))))
+      (values x1 y1 x2 y2))))
+
+(defmethod port-enable-sheet ((port wayland-port) (sheet mirrored-sheet-mixin))
+  (format t "port enable-sheet ~a ~a ~%" port sheet)
+  (alx:when-let ((mirror (sheet-direct-mirror sheet)))
+    (port-force-output port)))
+
+(defmethod port-disable-sheet ((port wayland-port) (sheet mirrored-sheet-mixin))
+  (format t "port disable-sheet ~a ~a ~%" port sheet)
+  (alx:when-let ((mirror (sheet-direct-mirror sheet)))
+    (wlc:wl-surface-attach (wayland-port-window port) nil 0 0)
+    (wlc:wl-surface-commit (wayland-port-window port))))
+
+(defmethod graft ((port wayland-port))
+  (first (port-grafts port)))
+
+;;; Grafts
+
+(defclass wayland-graft (graft)
+  ())
+
+(defmethod dispatch-event
+    ((sheet mirrored-sheet-mixin) (event window-configuration-event))
+  (format t "dispatch-event window-cfg-event: Not sure what should happen here...~%")
+  ;; I'm not sure if the is the right way. For now, let's handle it like an
+  ;; immediate-event-mixin
+  (handle-event sheet event))
+
+(defmethod handle-event
+    ((sheet wayland-graft) (event window-configuration-event))
+  ;; Is this where I should set the egl-window width and swap buffers?
+  ;; Should this be handled by the medium? or te mirror?
+  (alx:when-let ((mirror (sheet-mirror sheet)))
+    (format *debug-io* "handle-event window-cfg-event: top level reconfigure? Should I resize all the children?~%")
+    ;; QQQQ Should I use port-set-mirror-geometry or is there some other way
+    ;; to trigger it?
+    ;; (break)
```

Comment:
why configuration event is called on a graft? it should be called on a top level sheet. and the handle event is already implemented in the core. it calls internal function update-mirror-geometry.